examples/text_labels_and_annotations/text_rotation.py

```diff
@@ -0,0 +1,49 @@
+"""
+===========================
+Text rotation demonstration
+===========================
+
+The way matplotlib does text layout is counter-intuitive to some, so
+this example is designed to make it a little clearer.  The text is
+aligned by it's bounding box (the rectangular box that surrounds the
+ink rectangle).  The order of operations is rotation then
+alignment, rather than alignment then rotation.  Basically, the text
+is centered at your x,y location, rotated around this point, and then
+aligned according to the bounding box of the rotated text.
+
+So if you specify left, bottom alignment, the bottom left of the
+bounding box of the rotated text will be at the x,y coordinate of the
+text.
+
+But a picture is worth a thousand words!
+"""
+
+import matplotlib.pyplot as plt
+import numpy as np
+
+
+def addtext(ax, props):
+    ax.text(0.5, 0.5, 'text 0', props, rotation=0)
+    ax.text(1.5, 0.5, 'text 45', props, rotation=45)
+    ax.text(2.5, 0.5, 'text 135', props, rotation=135)
+    ax.text(3.5, 0.5, 'text 225', props, rotation=225)
+    ax.text(4.5, 0.5, 'text -45', props, rotation=-45)
+    ax.set_yticks([0, .5, 1])
+    ax.set_xlim(0, 5)
+    ax.grid(True)
```

Comment:
Would love to see some of the different rotation modes in this example similar to this: http://matplotlib.org/devdocs/examples/pylab_examples/demo_text_rotation_mode.html